src/service.ts

```diff
@@ -61,28 +73,25 @@ async function run(cancellationToken: CancellationToken) {
     }
   } catch (error) {
     if (error instanceof typescript.OperationCanceledException) {
-      return;
+      return undefined;
     }
 
     throw error;
   }
 
-  if (!cancellationToken.isCancellationRequested()) {
-    try {
-      process.send!({
-        diagnostics,
-        lints
-      });
-    } catch (e) {
-      // channel closed...
-      process.exit();
-    }
+  if (cancellationToken.isCancellationRequested()) {
+    return undefined;
   }
+
+  return {
+    diagnostics,
+    lints
+  };
 }
 
-process.on('message', message => {
-  run(CancellationToken.createFromJSON(typescript, message));
-});
+rpc.registerRpcHandler<RunPayload, RunResult>(RUN, message =>
+  run(CancellationToken.createFromJSON(typescript, message!))
```

Comment:
Shouldn't we check if message is not undefined?

Reply:
I think that's just a problem with the typings of worker-rpc - the lib is written for typescript 2.0 - and back then you could not correctly type that a method argument was there, or not - depending on generic.
This should in reality never be undefined.

But honestly, the check could not hurt, so I'll add it ;)